DQMOffline/Trigger/plugins/METplusTrackMonitor.cc

```diff
@@ -0,0 +1,461 @@
+   setMETitle(muonPtME_variableBinning_, "Muon p_{T} [GeV]", "events / [GeV]");
+
+   histname = "muonEta"; histtitle = "Muon eta";
+   bookME(ibooker, muonEtaME_, histname, histtitle, eta_binning_.nbins, eta_binning_.xmin, eta_binning_.xmax);
+   setMETitle(muonEtaME_, "Muon #eta", "events / 0.2");
+
+   histname = "deltaphi_muonmet"; histtitle = "dPhi(Muon, CaloMET)";
+   bookME(ibooker, deltaphimetmuonME_, histname, histtitle, phi_binning_.nbins, phi_binning_.xmin, phi_binning_.xmax);
+   setMETitle(deltaphimetmuonME_, "#Delta#phi (Muon, CaloMET)", "events / 0.1 rad");
+
+   histname = "mT_muonMet"; histtitle = "Transverse mass of muon-CaloMET system";
+   bookME(ibooker, mTmuonMetME_, histname, histtitle, pt_binning_.nbins, pt_binning_.xmin, pt_binning_.xmax);
+   setMETitle(mTmuonMetME_, "mT(Muon, CaloMET) [GeV]", "events / [GeV]");
+
+   histname = "muonPtVsLS"; histtitle = "Muon PT vs LS";
+   bookME(ibooker, muonPtVsLS_, histname, histtitle, ls_binning_.nbins, ls_binning_.xmin, ls_binning_.xmax, pt_binning_.xmin, pt_binning_.xmax);
+   setMETitle(muonPtVsLS_, "LS", "Muon p_{T} [GeV]");
+
+   // Initialize the GenericTriggerEventFlag
+  if(num_genTriggerEventFlag_ && num_genTriggerEventFlag_->on()) num_genTriggerEventFlag_->initRun(iRun, iSetup);
+  if(den_genTriggerEventFlag_ && den_genTriggerEventFlag_->on()) den_genTriggerEventFlag_->initRun(iRun, iSetup);
+
+ }
+
+ void METplusTrackMonitor::analyze(edm::Event const& iEvent, edm::EventSetup const& iSetup)
+ {
+
+   // Filter out events if Trigger Filtering is requested
+  if(den_genTriggerEventFlag_->on() && !den_genTriggerEventFlag_->accept(iEvent, iSetup)) return;
+
+  edm::Handle<reco::CaloMETCollection> metHandle;
+  iEvent.getByToken(metToken_, metHandle);
+  reco::CaloMET caloMet = metHandle->front();
+  if(!metSelection_(caloMet)) return;
+
+  float met = caloMet.pt();
+  float metPhi = caloMet.phi();
+
+  edm::Handle<reco::PFJetCollection> jetsHandle;
+  iEvent.getByToken(jetToken_, jetsHandle);
+  if(jetsHandle->size() < njets_) return;
+  std::vector<reco::PFJet> jets;
+  jets.clear();
+  for(auto const & j : *jetsHandle) {
+    if(jetSelection_(j)) jets.push_back(j);
+  }
+  if(jets.size() < njets_) return;
+  if(njets_ > 0 && jets.size() > 0 && fabs(jets[0].eta()) > leadJetEtaCut_) return;
+  float deltaphi_metjet1 = (jets.size() > 0) ? fabs(deltaPhi(caloMet.phi(), jets[0].phi())) : 10.0;
+
+  edm::Handle<reco::VertexCollection> primaryVertices;
+  iEvent.getByToken(vtxToken_, primaryVertices);
+  if(!primaryVertices->size()) return;
+  const reco::Vertex* pv = nullptr;
+  for(auto const& v: *primaryVertices) {
+    if (!vtxSelection_(v)) continue;
+    pv = &v;
+    break;
+  }
+  if(pv == nullptr) return;
+
+  edm::Handle<reco::MuonCollection> muonHandle;
+  iEvent.getByToken(muonToken_, muonHandle);
+  if(muonHandle->size() < nmuons_) return;
+  std::vector<reco::Muon> muons;
+  muons.clear();
+  for(auto const & m : *muonHandle) {
+    bool pass = m.isGlobalMuon() &&
```

Comment:
Yes; this is just reco::Muon::isTight and will be changed. Also I realize here that I haven't included muon isolation, which we want to require.